### PR TITLE
fix incorrect check verifying range proof

### PR DESCRIPTION
**Pull request by zhongxishen**
BG-74332

### CODE REVIEW

**Review by sachushaji — COMMENTED** [review]
Can you add a unit test for this fix which would fail if this fix was not made

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts [hunk 1]
```diff
@@ -258,7 +258,8 @@ export function verify(
   const modulusBytes = Math.floor((modulusBits + 7) / 8);
   const q = curve.order();
   const q3 = q ** BigInt(3);
-  if (proof.s1 == q3) {
+  // if s1 not <= q3 return false
```
Comment (zahin-mohammad) [inline modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts:261]: not sure we need this comment, the if statement is pretty clear

**Review by zahin-mohammad — COMMENTED** [review]
+1, on the test comment. Is this something we can easily test/reproduce?

File: modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts (hunk 1 above)
Comment (zhongxishen) [reply modules/sdk-core/src/account-lib/mpc/tss/ecdsa/rangeproof.ts]: removed

File: modules/sdk-core/test/unit/rangeproof.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+describe('MtA range proof', function () {
+  let switchPrime = false;
+  let safePrimeMock: sinon.SinonStub;
+
+  let paillierKeyPair: paillierBigint.KeyPair;
+  let ntilde: NTilde;
+
+  before('set up paillier and ntile', async function () {
+    safePrimeMock = sinon.stub(OpenSSL.prototype, 'generateSafePrime').callsFake(async (bitlength: number) => {
```
Comment (zhongxishen) [inline modules/sdk-core/test/unit/rangeproof.ts:16]: Stubbing `rangeProof.generateSafePrimes()` is more straightforward, but it doesn't work with node 14 for some reason.

**Review by sachushaji — APPROVED** [review]

File: modules/sdk-core/test/unit/rangeproof.ts (hunk 2 above)
Comment (zahin-mohammad) [reply modules/sdk-core/test/unit/rangeproof.ts:16]: `generateSafePrime` doesn't work on node 14 unit tests. Are you sure ur stubs are being called then?

**Review by zahin-mohammad — APPROVED** [review]

**Review by islamaminBitGo — APPROVED** [review]

**Pull request merged**